File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/team/lobby/respawn/RespawnLobby.java
package com.gmail.borlandlp.minigamesdtools.arena.team.lobby.respawn;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent;
import org.bukkit.entity.Player;

import java.util.Map;
import java.util.Set;

public interface RespawnLobby extends ArenaPhaseComponent {
    void update();
    void addPlayer(Player player);
    Set<Player> getReadyPlayersToRespawn();
    Map<Player, Long> getWaitingPlayers();
    void removePlayer(Player player);
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/team/lobby/respawn/RespawnLobby.kt
package com.gmail.borlandlp.minigamesdtools.arena.team.lobby.respawn

import com.gmail.borlandlp.minigamesdtools.arena.ArenaPhaseComponent
import org.bukkit.entity.Player

interface RespawnLobby : ArenaPhaseComponent {
    override fun update()
    fun addPlayer(player: Player)
    val readyPlayersToRespawn: Set<Player>
    val waitingPlayers: Map<Player, Long>
    fun removePlayer(player: Player)
}